### PR TITLE
4. the batch normalization

### PR DESCRIPTION
**Pull request by bensooraj**
## Model summary
```
==========================================================================================
Layer (type:depth-idx)                   Output Shape              Param #
==========================================================================================
NetBN                                    [1, 10]                   --
├─Sequential: 1-1                        [1, 8, 32, 32]            --
│    └─Conv2d: 2-1                       [1, 8, 32, 32]            224
│    └─BatchNorm2d: 2-2                  [1, 8, 32, 32]            16
│    └─ReLU: 2-3                         [1, 8, 32, 32]            --
├─Sequential: 1-2                        [1, 16, 32, 32]           --
│    └─Conv2d: 2-4                       [1, 16, 32, 32]           1,168
│    └─BatchNorm2d: 2-5                  [1, 16, 32, 32]           32
│    └─ReLU: 2-6                         [1, 16, 32, 32]           --
├─Sequential: 1-3                        [1, 8, 32, 32]            --
│    └─Conv2d: 2-7                       [1, 8, 32, 32]            136
│    └─BatchNorm2d: 2-8                  [1, 8, 32, 32]            16
│    └─ReLU: 2-9                         [1, 8, 32, 32]            --
├─MaxPool2d: 1-4                         [1, 8, 16, 16]            --
├─Sequential: 1-5                        [1, 16, 16, 16]           --
│    └─Conv2d: 2-10                      [1, 16, 16, 16]           1,168
│    └─BatchNorm2d: 2-11                 [1, 16, 16, 16]           32
│    └─ReLU: 2-12                        [1, 16, 16, 16]           --
├─Sequential: 1-6                        [1, 32, 16, 16]           --
│    └─Conv2d: 2-13                      [1, 32, 16, 16]           4,640
│    └─BatchNorm2d: 2-14                 [1, 32, 16, 16]           64
│    └─ReLU: 2-15                        [1, 32, 16, 16]           --
├─Sequential: 1-7                        [1, 64, 16, 16]           --
│    └─Conv2d: 2-16                      [1, 64, 16, 16]           18,496
│    └─BatchNorm2d: 2-17                 [1, 64, 16, 16]           128
│    └─ReLU: 2-18                        [1, 64, 16, 16]           --
├─Sequential: 1-8                        [1, 8, 16, 16]            --
│    └─Conv2d: 2-19                      [1, 8, 16, 16]            520
│    └─BatchNorm2d: 2-20                 [1, 8, 16, 16]            16
│    └─ReLU: 2-21                        [1, 8, 16, 16]            --
├─MaxPool2d: 1-9                         [1, 8, 8, 8]              --
├─Sequential: 1-10                       [1, 16, 8, 8]             --
│    └─Conv2d: 2-22                      [1, 16, 8, 8]             1,168
│    └─BatchNorm2d: 2-23                 [1, 16, 8, 8]             32
│    └─ReLU: 2-24                        [1, 16, 8, 8]             --
├─Sequential: 1-11                       [1, 32, 8, 8]             --
│    └─Conv2d: 2-25                      [1, 32, 8, 8]             4,640
│    └─BatchNorm2d: 2-26                 [1, 32, 8, 8]             64
│    └─ReLU: 2-27                        [1, 32, 8, 8]             --
├─Sequential: 1-12                       [1, 64, 8, 8]             --
│    └─Conv2d: 2-28                      [1, 64, 8, 8]             18,496
│    └─BatchNorm2d: 2-29                 [1, 64, 8, 8]             128
│    └─ReLU: 2-30                        [1, 64, 8, 8]             --
├─AvgPool2d: 1-13                        [1, 64, 1, 1]             --
├─Sequential: 1-14                       [1, 10, 1, 1]             --
│    └─Conv2d: 2-31                      [1, 10, 1, 1]             650
==========================================================================================
Total params: 51,834
Trainable params: 51,834
Non-trainable params: 0
Total mult-adds (Units.MEGABYTES): 9.48
==========================================================================================
Input size (MB): 0.01
Forward/backward pass size (MB): 1.13
Params size (MB): 0.21
Estimated Total Size (MB): 1.35
==========================================================================================
```

## Training and testing
**Significant improvement**
```
training accuracy: 66.80%
testing accuracy: 62.98%
number of epochs: 20
```
```
Accuracy for class:
- airplane is 68.6 %
- automobile is 80.5 %
- bird  is 39.8 %
- cat   is 33.3 %
- deer  is 59.7 %
- dog   is 53.2 %
- frog  is 72.3 %
- horse is 73.8 %
- ship  is 73.6 %
- truck is 75.0 %
```
Screenshot 2024-03-22 at 6 43 01 PM

## Comments
1. Increased the number of epochs from 10 to 20
2. Training and testing accuracies have improved significantly. Need to understand if there is over-fitting involved.
3. Number of trainable parameters have increased slightly from 51,306 to 51,834 (by +1.02%)

### CODE REVIEW

**Comment by bensooraj** [comment]
Screenshot 2024-03-22 at 10 10 27 PM

## Comments:
1. Some classes (such as `automobile`) are train better than the other (such as `cat`).
